CONTRIBUTING.md

```diff
@@ -190,6 +189,66 @@ When creating a pull-request (PR), it is **crucial** that the PR:
 - is organized in a cohesive sequence of commits, each representing a
   meaningful, logical and reviewable step.
 
+## Updating dependencies (`index-state`)
+
+Our Haskell packages come from two package repositories:
+- Hackage
+- [CHaP](https://github.com/input-output-hk/cardano-haskell-packages) (which is essentially another Hackage)
+
+The `index-state` of each repository is pinned to a particular time in
+`cabal.project`.  This tells Cabal to treat the repository as if it was
+the specified time, ensuring reproducibility.  If you want to use a package
+version from repository X which was added after the pinned index state
+time, you need to bump the index state for X.  This is not a big deal,
+since all it does is change what packages `cabal` considers to be available
+when doing solving, but it will change what package versions cabal picks
+for the plan, and so will likely result in significant recompilation, and
+potentially some breakage.  That typically just means that we need to fix
+the breakage (increasing the lower-bound on the problematic package if fix
+is not backward compatible), or delay that work and instead decrease the
```

Comment:
```suggestion
is not backwards compatible), or delay that work and instead decrease the
```

Reply:
Apparentely, that is fine: https://en.wikipedia.org/wiki/Backward_compatibility :smiley: